client.go

```diff
@@ -54,13 +55,16 @@ type Client struct {
 
 	// once initializes session
 	once sync.Once
+
+	// map of active subscriptions managed by this client
+	subscriptions map[uint32]Subscription
```

Comment:
I would guess that you need to guard this with a mutex and prevent direct access to it. This should also be a `map[uint32]*Subscription`. Same reasoning as before.

For formatting, I would also sort it above the `once`.

Reply:
yep. will do